ister_gui.py

```diff
@@ -1049,20 +1045,59 @@ def set_static_edits(self):
         urwid.connect_signal(self.dns_e, 'change', self._activate_button)
 
     def _activate_button(self, edit, new_text):
+        # on a reset, these need to be manually cleared
+        if self.reset:
+            for key in self.static_edits:
+                self.static_edits[key] = ''
+            self.reset = False
+
         self.static_edits[edit] = new_text
 
         # wait for all three required fields to be set updating the button
         # label with the new 'button' palette
         # `key` below is an object with a unique caption, so check against the
         # caption string.
+        # Additionally, validate the edit text for valid interface/ip format,
+        # make the caption green on a valid format.
         reqd_found = 0
         for key in self.static_edits:
             if 'Interface' in key.caption and self.static_edits[key]:
-                reqd_found += 1
+                if self.static_edits[key] in self.ifaceaddrs.keys():
+                    reqd_found += 1
+                    self.interface_e.set_caption(
+                            ('success', '{0:>20}'.format('Interface: ')))
+                else:
+                    self.interface_e.set_caption(
+                            '{0:>20}'.format('Interface: '))
+
             if 'IP address' in key.caption and self.static_edits[key]:
-                reqd_found += 1
+                try:
+                    ipaddress.ip_address(self.static_edits[key])
+                    reqd_found += 1
+                    self.static_ip_e.set_caption(
+                            ('success', '{0:>20}'.format('IP address: ')))
+                except:
+                    self.static_ip_e.set_caption(
+                            '{0:>20}'.format('IP address: '))
+
             if 'Gateway' in key.caption and self.static_edits[key]:
-                reqd_found += 1
+                try:
+                    ipaddress.ip_address(self.static_edits[key])
+                    reqd_found += 1
+                    self.gateway_e.set_caption(
+                            ('success', '{0:>20}'.format('Gateway: ')))
+                except:
+                    self.gateway_e.set_caption(('{0:>20}'.format('Gateway: ')))
+
+            if 'DNS' in key.caption and self.static_edits[key]:
+                try:
+                    ipaddress.ip_address(self.static_edits[key])
+                    self.dns_e.set_caption(
+                            ('success', '{0:>20}'.format('DNS (optional): ')))
+                except:
+                    self.dns_e.set_caption(
+                            '{0:>20}'.format('DNS (optional): '))
+                    reqd_found -= 1
```

Comment:
I am somewhat skeptical that the okay count is easier to understand than 4 different boolean variables but we can leave this as it is for now.